memif-attach/main.go

```diff
@@ -18,59 +18,102 @@ package main
 
 import (
 	"context"
+	"net"
 	"time"
-
+	// "io"
+	"encoding/binary"
+	"bufio"
+	"bytes"
 	"git.fd.io/govpp.git/api"
 	"github.com/edwarnicke/govpp/binapi/memif"
+	interfaces "github.com/edwarnicke/govpp/binapi/interface"
+	"github.com/edwarnicke/govpp/binapi/interface_types"
 	"github.com/edwarnicke/log"
 	"github.com/edwarnicke/vpphelper"
 )
 
+// HelloReplyMsg type
+type HelloMsg struct {
+	max_region int16 
+	max_m2s_ring int16 
+	max_s2m_ring int16
+	min_version int16 
+	max_version int16
+	max_log2_ring_size int8 
+}
+// UnmarshalBinary Function
+func (replyMsg *HelloMsg) UnmarshalBinary(data []byte) error {
+	buf := bytes.NewReader(data)
+	err := binary.Read(buf, binary.BigEndian, replyMsg)
```

Comment:
If you are not getting reasonable values with binary.BigEndian, try binary.LittleEndian instead.